pkg/registry/core/service/storage/rest_test.go

```diff
@@ -2037,39 +1394,27 @@ func TestServiceRegistryIPUpdate(t *testing.T) {
 	update.Spec.ClusterIP = testIP
 	update.Spec.ClusterIPs[0] = testIP
 
-	_, _, err := storage.Update(ctx, update.Name, rest.DefaultUpdatedObjectInfo(update), rest.ValidateAllObjectFunc, rest.ValidateAllObjectUpdateFunc, false, &metav1.UpdateOptions{})
+	_, _, err = storage.Update(ctx, update.Name, rest.DefaultUpdatedObjectInfo(update), rest.ValidateAllObjectFunc, rest.ValidateAllObjectUpdateFunc, false, &metav1.UpdateOptions{})
 	if err == nil || !errors.IsInvalid(err) {
 		t.Errorf("Unexpected error type: %v", err)
```

Comment:
not clear. should be separate if statements on for nil error and one for unexpected error.

Reply:
defer